docgen/generate/generate.go

```diff
@@ -0,0 +1,67 @@
+package generate
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+
+	google_protobuf "github.com/golang/protobuf/protoc-gen-go/descriptor"
+
+	"github.com/gunk/gunk/assets"
+	"github.com/gunk/gunk/docgen/parser"
+	"github.com/gunk/gunk/docgen/pot"
+)
+
+// Run generates a markdown file describing the API
+// and a messages.pot containing all sentences that need to be
+// translated.
+func Run(w io.Writer, f *google_protobuf.FileDescriptorProto) (pot.Builder, error) {
+	pb := pot.NewBuilder()
+
+	b, err := loadTemplate()
+	if err != nil {
+		return nil, err
+	}
+
+	tmpl := template.Must(template.New("api.md").
+		Funcs(template.FuncMap{
+			"GetText": func(txt string) string {
+				if txt != "" {
+					pb.AddTranslations([]string{txt})
+				}
+				return fmt.Sprintf("%s", txt)
+			},
+		}).
+		Parse(string(b)))
+
+	api, err := parser.ParseFile(f)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := tmpl.Execute(w, api); err != nil {
+		return nil, err
+	}
+	return pb, nil
+}
+
+func loadTemplate() ([]byte, error) {
+	file, err := assets.Assets.Open("api.md")
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	fi, err := file.Stat()
+	if err != nil {
+		return nil, err
+	}
+
+	f := make([]byte, fi.Size())
+	if _, err := file.Read(f); err != nil {
+		if err != io.EOF {
+			return nil, err
+		}
+	}
```

Comment:
Why not just use `ioutil.ReadAll`? I know there's another piece of code that does a manual read, but I think this is wrong.

For example, if you ask to read 500 bytes, the reader can tell you it read 250, and then read 250 the next time. That's valid, but it would break with this code here.

It also means you can get rid of all the unnecessary Stat stuff. You can simplify the other chunk of code too, if that's not too much for this PR :)